src/otx/core/model/module/base.py

```diff
@@ -133,12 +133,31 @@ def state_dict(self) -> dict[str, Any]:
         state_dict["meta_info"] = self.meta_info
         return state_dict
 
+    def _get_state_dict_from_ckpt(self, ckpt: dict[str, Any]) -> dict[str, Any]:
+        """Get the state_dict, supporting the backward compatibility."""
+
+        def load_from_prev_otx_ckpt(state_dict: dict[str, Any]) -> dict[str, Any]:
+            """Attach the model.model prefix to load without problem."""
+            for key in list(state_dict.keys()):
+                value = state_dict.pop(key)
+                new_key = "model.model." + key
+                state_dict[new_key] = value
+            return state_dict
+
+        # TODO(sungmanc): need to consider OTXv0.x and openvino pretrained model # noqa: TD003
+        if "model" in ckpt and ckpt["VERSION"] == 1:
+            model_state_dict = ckpt["model"]["state_dict"]
+            return load_from_prev_otx_ckpt(model_state_dict)
+        return ckpt["state_dict"]
+
     def load_state_dict(self, state_dict: dict[str, Any], *args, **kwargs) -> None:
         """Load state dictionary from checkpoint state dictionary.
 
         If checkpoint's meta_info and OTXLitModule's meta_info are different,
         load_state_pre_hook for smart weight loading will be registered.
         """
+        state_dict = self._get_state_dict_from_ckpt(state_dict)
```

Comment:
```suggestion
        if ckpt["VERSION"] == 1 and "model" in state_dict:
            model_state_dict = state_dict["model"]["state_dict"]
            state_dict = _load_from_otx_1_x_ckpt(model_state_dict)
```

Reply:
Thanks for the suggestion, however, OTX2.0 checkpoint doesn't have `VERSION` key. We might have another option:  change the `state_dict` function to save the `VERSION` info at the OTX2.0.

Reply:
I still cannot understand why `_get_state_dict_from_ckpt()` should exist. It does something if `"model" in ckpt and ckpt["VERSION"] == 1`. It means that it exists for doing something for `version==1`. We cannot infer anything about its behavior from its naming.

Reply:
I get your point, thanks. https://github.com/openvinotoolkit/training_extensions/pull/2812/commits/c1a146cc4c63d768a4adca7b670fce1d73747a5c